pykechain/models/widgets/widget.py

```diff
@@ -195,11 +191,68 @@ def update_associations(self, readable_models=None, writable_models=None, **kwar
         :raises APIError: when the associations could not be changed
         :raise IllegalArgumentError: when the list is not of the right type
         """
-        self._client.update_widget_associations(widget=self, readable_models=readable_models,
-                                                writable_models=writable_models, **kwargs)
+        self._client.update_widget_associations(
+            widget=self,
+            readable_models=readable_models,
+            writable_models=writable_models,
+            **kwargs
+        )
+
+    def set_associations(
+            self,
+            readable_models: Optional[List] = None,
+            writable_models: Optional[List] = None,
+            **kwargs
+    ) -> None:
+        """
+        Set associations on this widget.
+
+        This is an absolute list of associations. If you provide No models, than the associations are cleared.
+
+        Alternatively you may use `inputs` or `outputs` as a alias to `readable_models` and `writable_models`
+        respectively.
+
+        :param readable_models: list of property models (of :class:`Property` or property_ids (uuids) that has
+                                read rights (alias = inputs)
+        :type readable_models: List[Property] or List[UUID] or None
+        :param writable_models: list of property models (of :class:`Property` or property_ids (uuids) that has
+                                write rights (alias = outputs)
+        :type writable_models: List[Property] or List[UUID] or None
+        :param kwargs: additional keyword arguments to be passed into the API call as param.
+        :return: None
+        :raises APIError: when the associations could not be set
+        :raise IllegalArgumentError: when the list is not of the right type
+        """
+        self._client.set_widget_associations(
+            widget=self,
+            readable_models=readable_models,
+            writable_models=writable_models,
+            **kwargs
+        )
+
+    def remove_associations(
+            self,
+            models: List[Union['Property', Text]],
+            **kwargs
+    ) -> None:
+        """
+        Remove associated properties from the widget.
+
+        :param models: list of Properties or their uuids
+        :return: None
+        """
+        self._client.remove_widget_associations(
```

Comment:
if a grid has all associations removed? What happens. The grid is not configured. It should have at least a single model association. 

I guess we can come into a state where we have a un-associated widget. Unsure how to deal with that.

Reply:
Have an argument "remove_widget", default=True? That will remove the widget if its associations are removed, but the developer can potentially overwrite that if he wants to use said widget later.